src/cli/cli_commissioner.cpp

```diff
@@ -88,6 +88,46 @@ template <> otError Commissioner::Process<Cmd("joiner")>(Arg aArgs[])
     const otExtAddress *addrPtr = nullptr;
     otJoinerDiscerner   discerner;
 
+    if (aArgs[0] == "list")
+    {
+        uint16_t     iter = 0;
+        otJoinerInfo joinerInfo;
+
+        OutputLine("| ID                   | PSKd                           | Expiration |");
+        OutputLine("+----------------------+--------------------------------+------------+");
+
+        while (otCommissionerGetNextJoinerInfo(GetInstancePtr(), &iter, &joinerInfo) == OT_ERROR_NONE)
+        {
+            switch (joinerInfo.mType)
+            {
+            case OT_JOINER_INFO_TYPE_ANY:
+                OutputFormat("|%22s", "*");
+                break;
+
+            case OT_JOINER_INFO_TYPE_EUI64:
+                OutputFormat("| ");
+                OutputBytes(joinerInfo.mSharedId.mEui64.m8, OT_EXT_ADDRESS_SIZE);
+                OutputFormat("     ");
+                break;
+
+            case OT_JOINER_INFO_TYPE_DISCERNER:
+            {
+                OutputFormat("| 0x%016llx/%2u", static_cast<unsigned long long>(joinerInfo.mSharedId.mDiscerner.mValue),
+                             joinerInfo.mSharedId.mDiscerner.mLength);
+            }
+            break;
+            }
+
+            OutputFormat("|%32s", joinerInfo.mPskd.m8);
+            OutputFormat("| %10d", joinerInfo.mExpirationTime);
+            OutputLine(" |");
```

Comment:
We can combine into a single `OutputLine("| %32s | %10d |", ...)`?

Reply:
Done.